packages/flutter/lib/src/services/message_codec.dart

```diff
@@ -26,6 +26,62 @@ abstract class MessageCodec<T> {
   T decodeMessage(ByteData message);
 }
 
+/// An command object representing the invocation of a named method.
+class MethodCall {
+  final String method;
+  final dynamic arguments;
+  
+  /// Creates a [MethodCall] representing the invocation of [method] with the
+  /// specified [arguments].
+  MethodCall(this.method, [this.arguments]) {
+    assert(method != null);
+  }
+  
+  @override
+  bool operator== (dynamic o) {
+    if (o is! MethodCall)
+      return false;
+    final MethodCall other = o;
+    return method == other.method && _deepEquals(arguments, other.arguments);
+  }
+  
+  bool _deepEquals(dynamic a, dynamic b) {
+    if (a == b)
+      return true;
+    if (a is List)
+      return b is List && _deepEqualsList(a, b);
+    if (a is Map)
+      return b is Map && _deepEqualsMap(a, b);
+    return false;
+  }
+  
+  bool _deepEqualsList(List<dynamic> a, List<dynamic> b) {
+    if (a.length != b.length)
+      return false;
+    for (int i = 0; i < a.length; i++) {
+      if (!_deepEquals(a[i], b[i]))
+        return false;
+    }
+    return true;
+  }
+
+  bool _deepEqualsMap(Map<dynamic, dynamic> a, Map<dynamic, dynamic> b) {
+    if (a.length != b.length)
+      return false;
+    for (dynamic key in a.keys) {
+      if (!b.containsKey(key) || !_deepEquals(a[key], b[key]))
+        return false;
+    }
+    return true;
+  }
+  
+  @override
+  int get hashCode => method.hashCode ^ (arguments?.hashCode ?? 0);
```

Comment:
also if possible put == and hashCode together so it's obvious that they're equivalent